contracts/reputationMiningCycle/ReputationMiningCycle.sol

```diff
@@ -69,7 +69,7 @@ contract ReputationMiningCycle is ReputationMiningCycleCommon {
     }
   }
 
-  uint256 constant X = UINT256_MAX / MINING_WINDOW_SIZE;
+  uint256 constant X = UINT256_MAX / (MINING_WINDOW_SIZE - ALL_ENTRIES_ALLOWED_END_OF_WINDOW);
```

Comment:
I would prefer this expressed as a fraction and calculated using `wmul`, which avoids the possible (albeit unlikely) case of the all-entry window being larger than the mining window.

Reply:
The scenario you're worried about here is us changing these constants incorrectly down the line?

Reply:
For what it's worth, a wad would also have a huge amount of invalid space in its range (anything bigger than 1). 

I don't think I can do what you're proposing without using storage slots and initialising them on deployment, or recalculating every time we use functions that need this constant, as we can only use solidity built-ins for defining constants.

Reply:
Alright, so be it.